src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GlobalExceptionHandler.kt

```diff
@@ -150,6 +152,19 @@ class GlobalExceptionHandler(
     return populateErrorResponseAndHandleExceptionInternal(e, BAD_REQUEST, request)
   }
 
+  /**
+   * Exception handler to return a 503 Service Unavailable ErrorResponse, specifically for a DataAccessException.
+   */
+  @ExceptionHandler(DataAccessException::class)
+  fun handleDataAccessException(
+    e: DataAccessException,
+    request: WebRequest,
+  ): ResponseEntity<Any>? {
+    log.error("Unexpected database exception", e)
+    request.setAttribute(ERROR_MESSAGE, "Service unavailable", SCOPE_REQUEST)
```

Comment:
A nice vague error message that will get rendered out in the API `ErrorResponse`, but the exception itself is logged so we can still find and investigate it 👍